client/landing/gutenboarding/onboarding-block/vertical-select/index.tsx

```diff
@@ -139,6 +140,10 @@ const VerticalSelect: FunctionComponent< StepProps > = ( {
 		}
 	}, [ isActive, inputRef ] );
 
+	useEffect( () => {
+		recordTracksPageViewWithPageParams( '/gutenboarding/about' );
+	}, [] );
```

Comment:
Just a quick comment for now: I wonder if this could be done in a general way rather than having to call `recordTracksPageViewWithPageParams` on mount of each component. Plus it's susceptible to breaking when we make changes to the route.

I was thinking about listening to route updates instead of component mounts e.g. adding something like this to the `<Gutenboard>` component:

```typescript
import { useParams, useHistory } from 'react-router-dom';

const { location: { pathname } } = useHistory();
useEffect( () => {
	recordTracksPageViewWithPageParams( `/gutenboarding${pathname}` );
}, [ pathname ] );
```

Adding it to `gutenboard.tsx` instead of `index.tsx` means it will only be triggered by routes that we actually support i.e. no `/gutenboarding/abcxyz` track events.

Another thing it will do differently is send the locale slug as part of the event too. I'm not sure whether that's how `recordTracksPageViewWithPageParams` is supposed to be used, but I assume so.

What do you think?

Reply:
Just checked the old signup framework, and yes it's supposed to include the language slug.

Reply:
nice, that looks good, I'll add that code and test it, thanks.